### PR TITLE
feat: courier templates fs support

### PR DESCRIPTION
**Pull request by Benehiko**
Adds `embed.FS` support to the templates.

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

File: courier/template/load_template.go [hunk 1]
```diff
@@ -105,23 +129,41 @@ func loadTemplate(osdir, name, pattern string, html bool) (Template, error) {
...
 	var b bytes.Buffer
 	if err := t.Execute(&b, model); err != nil {
 		return "", err
 	}
 	return b.String(), nil
 }
 
-func LoadHTMLTemplate(osdir, name, pattern string, model interface{}) (string, error) {
-	t, err := loadTemplate(osdir, name, pattern, true)
+func LoadHTMLTemplate(osdir, name, pattern string, model interface{}, fs ...embed.FS) (string, error) {
```
Comment (aeneasr) [inline courier/template/load_template.go:153]: It looks like we will only use the first embedFS argument. So maybe just make this a regular arg?

**Review by aeneasr — CHANGES_REQUESTED** [review]

File: courier/template/load_template.go (hunk 1 above)
Comment (Benehiko) [reply courier/template/load_template.go:153]: but then it won't be optional anymore. we will then require the current functions to pass a nil i guess

Comment (Benehiko) [reply courier/template/load_template.go]: I see, okay, I will implement option 2.

Comment (aeneasr) [reply courier/template/load_template.go]: Thank you @Benehiko  :)

**Comment by Benehiko** [comment]
Just to make sure I'm not adding any regressions. What is the purpose of the `pattern` parameter in this case? To me, it seems strange since if `name` doesn't match, `pattern` doesn't matter. So we know the file exists through its `name`. But then we use the `pattern` (if it exists) to call up the file instead. 

https://github.com/ory/kratos/blob/master/courier/template/load_template.go#L80-L87

**Review by zepatrik — APPROVED** [review]

**Comment by aeneasr** [comment]
Great stuff, thank you @Benehiko !

**Pull request merged**